### DISCUSSION TITLE
Questions for LC Net Dev

### DISCUSSION THREAD

**Discussion by CECSpecialistI** [crystalyragui/MARC2RDA#388] [General]
We're thinking of inviting someone from LC, such as Kevin Ford, to one of our meetings to answer some questions we have about the way metadata is modeled at id.loc.gov. This discussion is a place where we can formulate those questions together. @lake44me @GordonDunsire @gerontakos @AdamSchiff @briesenberg07 @SitaKB @szapoun @pan-zhuo @cwarc @JianPLee @junghaelee

**Comment by briesenberg07** [comment]
From the meeting [notes for 02-15](https://github.com/uwlib-cams/MARC2RDA/wiki/2023-Meeting-Minutes#february-15-2023-800am---930am-pst):

> Are $0 identifiers/IRIs for authorized access points/names/labels? For datasets to which the IRI dereferences?

I don't understand the question. Do I just misunderstand, or does it need to be clarified? 

Instead of asking about "$0 IRIs," which are entered by humans and could be anything, do we want to ask about a certain class or classes of id.loc IRIs? For example:

- Questions about LoC resources typed as `madsrdf:Authority` + `madsrdf:PersonalName` + `skos:Concept` (id.loc.gov/authorities/names/[...] IRIs)
- Questions about LoC resources typed as `bf:Person` + `madsrdf:RWO` + `foaf:Person` (id.loc.gov/rwo/agents/[...] IRIs)?

If I'm correct and our questions are about the nature of id.loc RDF, I'm not sure what we would expect to learn from someone at LC beyond what we can take from looking at triples obtained from their Linked Data Service (what they might tell us beyond how the data describes itself). 

*I may be misunderstanding entirely--perhaps our questions are more about MARC subfield $0 than about id.loc data-modeling?*

**Reply by GordonDunsire** [reply]
I think any meaningful questions are about data modelling.

To summarise the issue(s) I have raised, take the case of the entry in the Name Authority File for "Mirna Willer".

The HTML says:

URI: https://id.loc.gov/authorities/names/no2009154799
Additional information: https://id.loc.gov/rwo/agents/no2009154799

The n-triple for the "additional information" linking statement is:

<http://id.loc.gov/authorities/names/no2009154799> <http://www.loc.gov/mads/rdf/v1#identifiesRWO> <http://id.loc.gov/rwo/agents/no2009154799>

This implies, via a subproperty relationship:

<http://id.loc.gov/authorities/names/no2009154799> <http://xmlns.com/foaf/0.1/focus> <http://id.loc.gov/rwo/agents/no2009154799>

The de-referenced graph of the "names" URI embeds the complete de-referenced graph of the "rwo" URI.

Minor question: why embed the whole "rwo" graph, rather than provide the linking triple and leave it to applications to de-reference the "rwo" URI to obtain the graph?

The following types/classes/entities are explicitly declared for the "name" URI:

- <http://www.loc.gov/mads/rdf/v1#Authority>
- <http://www.loc.gov/mads/rdf/v1#PersonalName>
- <http://www.w3.org/2004/02/skos/core#Concept>

The following types/classes/entities are explicitly declared for the "rwo" URI:

- <http://www.loc.gov/mads/rdf/v1#RWO>
- <http://id.loc.gov/ontologies/bibframe/Person>
- <http://xmlns.com/foaf/0.1/Person>

Another n-triple says:

<http://id.loc.gov/authorities/names/no2009154799> <http://www.loc.gov/mads/rdf/v1#hasCloseExternalAuthority> <http://www.wikidata.org/entity/Q55449547>

The type/class/entity for the "wikidata" URI is explicitly declared as <http://www.wikidata.org/entity/Q5> "human". This entity/class is a subclass of "natural person" < "legal person" < "agent". [Seems that RDA Person is equivalent to wikidata Natural Person; the "human" subclass has little semantic distinction.]

<https://www.wikidata.org/wiki/Q154954> "natural person" is declared in wikidata to be an instance of (not subclass of) "legal concept", which is a subclass of <https://www.wikidata.org/wiki/Q151885> "concept"

But the range of mads:hasCloseExternalAuthority entails that the wikidata URI is an instance of mads:Authority.

Question: is the wikidata URI an instance of id.loc.gov authority and id.loc.gov real-world object, simulaneously?

As far as I can see*** in [Wikidata Help](https://www.wikidata.org/wiki/Help:Basic_membership_properties), in wikidata "instance of" is not transitive: a _instance of_ b _instance of_ c does not entail a _instance of_ c. So a URI that is an instance of "human" is not an instance of "concept" just because "human" is an instance of "concept"; the instance URI references an individual, but "human" references a class: the class "human" is an instance of the class "concept".

On the other hand, "subclass of" is transitive: a _subclass of_ b _subclass of c_ entails a _subclass of_ c; and "instance of + subclass of" is quasi-transitive: a _instance of_ b _subclass of_ c entails a _instance of_ c.

*** I am not at all familiar with Wikidata's data model, so I may be misunderstanding this.

**Comment by briesenberg07** [comment]
🐇🕳️ **ASIDE**
The property foaf:focus (with IRI given here as http://xmlns.com/foaf/0.1/focus), pops up in this discussion (because per MADS/RDF documentation, [madsrdf:identifiesRWO](https://id.loc.gov/ontologies/madsrdf/v1.html#identifiesRWO) is a subproperty of it) and in the [discussion on IRIs/RWOs](https://github.com/uwlib-cams/MARC2RDA/discussions/384#discussioncomment-4809527), ~~but I haven't been able to find a definition in the [FOAF Vocabulary Specification](http://xmlns.com/foaf/0.1/) or on the web.~~

Nevermind--apologies, I see Gordon's links to [info from a mailing list](https://lists.w3.org/Archives/Public/public-esw-thes/2010Aug/0002.html), etc.

**Comment by CECSpecialistI** [comment]
We'll send an email asking LC about Wikidata as authorities rather than RWO's

**Reply by CECSpecialistI** [reply]
@briesenberg07 and @gerontakos will look more closely at bibframe data and work on an email

**Reply by briesenberg07** [reply]
I wrote up one of the sets of questions we discussed**, with reference to some test data.  

\*\***Why are LC *names* resources (and not *agents* resources) linked to persons in Wikidata? Etc.**  

> [Data-modeling questions for LC Net Dev: Linking LCNAF entities to Wikidata items](https://gist.github.com/briesenberg07/127532b2b1088141a39b8d5901205acd#data-modeling-questions-for-lc-net-dev)

Please take a look if you like and pass along any feedback (My last SPARQL query could be improved...).

I can send this along to Kevin Ford at LC when ready.

**Reply by GordonDunsire** [reply]
@briesenberg07: The write-up looks ok to me; my sparql is too rusty to comment!

**Reply by CECSpecialistI** [reply]
Looks good to me! I trust that your SPARQL skills are better than mine, so won't try to improve on your query. Very clearly-stated questions and supporting documentation! Thank you Benjamin :)

**Comment by gerontakos** [comment]
The questions are good. I didn't check the queries. I hope you plan to send data retrieved rather than queries to run in some python environment. You are sending data that exemplify the problems, right?

  --Theo
________________________________

**Reply by briesenberg07** [reply]
Thank you @gerontakos -- to answer your comment and question:

> I hope you plan to send data retrieved rather than queries to run in some python environment.

The data and gist I thought of as more for me than for LC, who of course have the data. I wanted to be sure that what I thought was happening was happening, at least in a small sample set. 

I'll most likely say something like 'these observations are based on a small test set of 50 LCNAF mads:Authority entity descriptions'. It would be easy enough to send an IRI list along with the email.

> You are sending data that exemplify the problems, right?

The 50 sample descriptions do exemplify the things we think are problems.

**Comment by GordonDunsire** [comment]
These issues may be related to a short BF discussion thread from late last year on [LoC relators as properties (Nov 2023)](https://listserv.loc.gov/cgi-bin/wa?A1=ind2211&L=BIBFRAME#6) and [LoC relators as properties (Dec 2023)](https://listserv.loc.gov/cgi-bin/wa?A1=ind2212&L=BIBFRAME#1). [It's odd that the thread is in disconnected parts ...]

**Reply by briesenberg07** [reply]
Thank you @GordonDunsire !

**Comment by briesenberg07** [comment]
***Also, received a reply from Kevin Ford at LOC yesterday. Kevin kindly let me know I could follow up if any lingering questions remain. Posting our stated questions and Kevin's reply below:***

# QUESTIONS

## Why are id.loc.gov names resources used to link to Wikidata items, not id.loc.gov agents (rwo) resources?
 
- Names entities link to WD items which are instances of (wdt:P31) human (wd:Q5).
- Agents resources, which are typed as bf:Person, madsrdf:RWO, and foaf:Person seem better aligned than names resources typed as madsrdf:Authority.
 
## When creating links from names entities to Wikidata items, why are skos:closeMatch and madsrdf:hasCloseExternalAuthority used much more frequently than madsrdf:identifiesRWO to link to WD items?
 
- Documentation for skos:closeMatch and madsrdf:hasCloseExternalAuthority doesn't specify range, but in both cases users of data might assume the objects of triples to be concepts or authorities, rather than real-world objects like humans.
 
## Why are Wikidata items typed as skos:Concepts and madsrdf:Authority in the LCNAF entity-description files?
 
-   I'm not aware of any modeling which defines human (wd:Q5) as disjoint with these classes, but at face value these assertions are questionable given the Wikidata modeling for human and commonsense understanding.

# Response from Kevin Ford 2023-04-26

For better or for worse, when we started adding links from resources in ID to Wikidata resources, our focus was merely on getting links in place and not delving into ontological matters of what is an RWO, etc.  Putting aside how wikidata defines things ontologically, Wikidata entities are often seen as authorities, or authoritative these days, and so aligning them with the Authority seemed defensible.  Thus there was an element of expediency (because it was a handy solution not just for Names but also LCSH and LCFGT etc, which were part of the matching project) with some light consideration given to the semantics.  Ergo, for wikidata links that *we* determined – meaning the processes we have that run weekly (at least) to match NACO resources with Wikidata entities - we have made the assertion X closeAuthority WD.  That’s why the bulk of the relationships you see in your SPARQL queries take this structure.  We make the match and store the data to the side as X closeAuthority WD.   (The fact that the WD Entities are being associated with madsrdf:Authority and receiving that RDF type is a byproduct of this, of being hasCloseMatchingAuthority.)
 
The smaller number of X identifiesRWO WD are actually coming from the 024, i.e. the MARC source data.   That’s what is happening in this particularly rich example:
 
[https://id.loc.gov/authorities/names/n79076619.rdf](https://urldefense.com/v3/__https:/id.loc.gov/authorities/names/n79076619.rdf__;!!K-Hz7m0Vt54!nrlQTHmqu4TXfKUDAFga5IUvyplVsRiqgCD6B9r48J1bnP28k6tX1WEL2Ngan8ULpUsySuIdydIM-hWgOg$)  (MADS/RDF + SKOS)
[https://id.loc.gov/authorities/names/n79076619.marcxml.xml](https://urldefense.com/v3/__https:/id.loc.gov/authorities/names/n79076619.marcxml.xml__;!!K-Hz7m0Vt54!nrlQTHmqu4TXfKUDAFga5IUvyplVsRiqgCD6B9r48J1bnP28k6tX1WEL2Ngan8ULpUsySuIdydLUM_kDAA$)   (MARC/XML)
 
Our handling of the 024 was independent of the matching project (can’t recall if it came before or after, and probably does not matter).  But whomever did the 024 was different from the person who did the matching (and who led the charge of making those triples), so you have hit upon an inconsistency the crept in there because of time and different hands.
 
I’ll put this on the list of things to disentangle.  What is happening with the 024 is better for WD entities, at least for Names.  It probably won’t happen anytime soon, however, just to warn you.
 
Thanks for pointing this out and let me know if I’ve not answered one of your questions.
